File: evidence/generated-systems/golang/manufacturing-on-golang/manufacturing-on-golang/internal/model/Asset.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// Asset Declaration
//==============================================================
type Asset struct {
    gorm.Model
     AssetTag                                    string
    AssetName                                    string
    CommissioningDate                                                            time.Time
    PlantId         *uint
    Plant           *Plant `gorm:"foreignKey:PlantId"`
    WorkCenterId         *uint
    WorkCenter           *WorkCenter `gorm:"foreignKey:WorkCenterId"`
     MaintenanceOrders           []MaintenanceOrder `gorm:"foreignKey:MaintenanceOrdersFromAssetId"`
     MaintenancePlans           []MaintenancePlan `gorm:"foreignKey:MaintenancePlansFromAssetId"`
    AssetStatus                      AssetStatus

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/manufacturing-on-rails/manufacturingonrails/app/models/asset.rb
class Asset < ApplicationRecord
  enum AssetStatus: [:Commissioned, :Available, :InMaintenance, :Down, :Retired]


  has_many :Plant, class_name: 'Plant'
  has_many :WorkCenter, class_name: 'WorkCenter'
  has_many :MaintenanceOrders, class_name: 'MaintenanceOrder'
  has_many :MaintenancePlans, class_name: 'MaintenancePlan'

end
